backend/items/collection/filter.go

```diff
@@ -0,0 +1,54 @@
+package collection
+
+import (
+	"database/sql"
+	"encoding/json"
+	"github.com/bcc-code/brunstadtv/backend/common"
+	"github.com/bcc-code/brunstadtv/backend/jsonlogic"
+	"github.com/lib/pq"
+)
+
+func itemIdsFromRows(rows *sql.Rows) []int {
+	var ids []int
+
+	for rows.Next() {
+		var id int
+		_ = rows.Scan(&id)
+		ids = append(ids, id)
+	}
+
+	_ = rows.Close()
+	return ids
+}
+
+// GetItemIDsForFilter returns an array of ids for the collection
+func GetItemIDsForFilter(db *sql.DB, collection string, f common.Filter) ([]int, error) {
+	if f.Filter == nil {
+		return nil, nil
+	}
+
+	var filterObject map[string]any
+	_ = json.Unmarshal(f.Filter, &filterObject)
+	filterString := jsonlogic.GetSQLStringFromFilter(filterObject)
+
+	var orderByString string
+	if f.SortBy != "" {
+		orderByString = " ORDER BY " + pq.QuoteIdentifier(f.SortBy)
+	}
+	if orderByString != "" && f.SortByDirection != "" {
+		switch f.SortByDirection {
+		case "desc":
+			orderByString += " DESC"
+		case "asc":
+			orderByString += " ASC"
+		}
+	}
+
+	queryString := "SELECT id FROM " + pq.QuoteIdentifier(collection) + " WHERE " + filterString + orderByString
+
+	rows, err := db.Query(queryString)
```

Comment:
I'm just wondering if we should add a debug log here for previews (if easy) so we could easily get see what SQL we actually executed if we need to debug something.

Reply:
Will log now, if preview is set in context (if that's sufficient)